### PR TITLE
avm2: Clamp ByteArray.position when ByteArray.length is set

### PR DESCRIPTION
**Pull request by Aaron1011**
Note that ByteArray.position can manually be set out-of-range afterwards.

### CODE REVIEW

**Comment by Aaron1011** [comment]
Fixes https://github.com/ruffle-rs/ruffle/pull/12244

**Review by Dinnerbone — APPROVED** [review]

File: tests/tests/swfs/avm2/bytearray/Test.as [hunk 1]
```diff
@@ -105,4 +105,10 @@ trace(ba1.readBoolean());
...
 trace(ba1.readUnsignedInt());
-trace(ba1.readInt());
\ No newline at end of file
+trace(ba1.readInt());
+
+var ba2 = new ByteArray();
+ba2.position = 100;
+trace("ba2.position: " + ba2.position);
+ba2.length = 2;
+trace("ba2.position: " + ba2.length);
```
Comment (evilpie) [inline tests/tests/swfs/avm2/bytearray/Test.as:114]: This should be ba2.position

Comment (torokati44) [reply tests/tests/swfs/avm2/bytearray/Test.as:114]: Or perhaps the printed text should be `"ba2.length: "`?

Comment (Aaron1011) [reply tests/tests/swfs/avm2/bytearray/Test.as]: Good catch - I changed it to access `ba2.position`

Comment (torokati44) [reply tests/tests/swfs/avm2/bytearray/Test.as]: What's the point of tracing position again after only changing length? My impression was that the point of this last trace is to prove that length can be changed. But I don't mind either way.

Comment (Aaron1011) [reply tests/tests/swfs/avm2/bytearray/Test.as]: This PR clamps the position when `length` is changed - note that the two `ba2.position` values are different.

Comment (torokati44) [reply tests/tests/swfs/avm2/bytearray/Test.as]: Ah, alright, sorry for the trouble.

**Pull request merged**